src/account/Ed25519Account.ts

```diff
@@ -75,13 +77,17 @@ export class Ed25519Account implements Account {
     return this.publicKey.verifySignature(args);
   }
 
-  signWithAuthenticator(message: HexInput) {
-    const signature = this.privateKey.sign(message);
+  signWithAuthenticator(transaction: AnyRawTransaction) {
```

Comment:
why this change?

Reply:
Signing with an authenticator really makes sense if it's an AnyRawTransaction.

And I think the signing message generation logic should be explicit.  The motivation for this change is that Keyless will do it a bit differently.